firedrake/mg/kernels.py

```diff
@@ -28,10 +28,53 @@
 from tsfc.parameters import default_parameters
 from tsfc.finatinterface import create_element
 from finat.quadrature import make_quadrature
-from firedrake.pointquery_utils import dX_norm_square, X_isub_dX, init_X, inside_check, is_affine, compute_celldist
+from firedrake.pointquery_utils import dX_norm_square, X_isub_dX, init_X, inside_check, is_affine
 from firedrake.pointquery_utils import to_reference_coords_newton_step as to_reference_coords_newton_step_body
 
 
+def compute_celldist_mg(fiat_cell, X="X", celldist="celldist"):
+    """Print C code to find the maximum of an input pointer string ``X`` which
+    is then multiplied by -1 with the result stored in ``celldist``.
+
+    If the ``fiat_cell`` is one dimensional then this doesn't find the maximum,
+    instead it simply multiplies the first entry of ``X`` by -1 and stores it
+    in ``celldist``.
+
+    **WARNING**: This function is specialised for use in multigrid kernels and
+    doesn't really compute a cell distance in any usual sense.
```

Comment:
I think the incorrect computation is just a bug, what happens if you fix it?

Reply:
To fix this we need to use the new `celldist_l1_c_expr` function in `pointquery_utils.py` and update the kernels themselves to work more like the new `locate.c` - I can have a go some time in a separate PR

Reply:
Draft PR running tests here https://github.com/firedrakeproject/firedrake/pull/2759